admin-dev/themes/new-theme/js/pages/product/edit/virtual-product-manager.ts

```diff
@@ -76,4 +77,27 @@ export default class VirtualProductManager {
   private showContent(): void {
     this.$fileContentContainer.removeClass('d-none');
   }
+
+  private fillVirtualProductNameField(): void {
+    $(ProductMap.virtualProduct.fileUploadField).on('change', () => {
+      let fullPath = $(ProductMap.virtualProduct.fileUploadField).val()!;
+
+      if (Array.isArray(fullPath)) {
+        // Handle the error with the most appropriate way, here an error is just thrown
+        throw new Error('Full path of the file is not supposed to be an array');
+      } else {
+        // Convert to string in any other case
+        fullPath = fullPath.toString();
+      }
+
+      // get file name from full path
+      const startIndex = (fullPath.indexOf('\\') >= 0 ? fullPath.lastIndexOf('\\') : fullPath.lastIndexOf('/'));
+      let filename = fullPath.substring(startIndex);
+
+      if (filename.indexOf('\\') === 0 || filename.indexOf('/') === 0) {
+        filename = filename.substring(1);
+      }
+      $(ProductMap.virtualProduct.fileNameField).val(filename);
```

Comment:
Maybe this whole prefill should only be done if `$(ProductMap.virtualProduct.fileNameField).val()` is empty If you already typed the name it will be erased on file selection this way

Reply:
nice catch 👍